stream/src/main/java/org/apache/apex/malhar/stream/api/impl/ApexStreamImpl.java

```diff
@@ -0,0 +1,493 @@
+        if (field.getType() == Attribute.class) {
+          DAG_ATTRIBUTES.add((Attribute)field.get(Context.DAGContext.class));
+        }
+      }
+    } catch (IllegalAccessException e) {
+      //Ignore here
+    }
+
+    INPUT_ATTRIBUTES.add(Context.PortContext.PARTITION_PARALLEL);
+    INPUT_ATTRIBUTES.add(Context.PortContext.AUTO_RECORD);
+    INPUT_ATTRIBUTES.add(Context.PortContext.STREAM_CODEC);
+    INPUT_ATTRIBUTES.add(Context.PortContext.TUPLE_CLASS);
+
+
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.QUEUE_CAPACITY);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.BUFFER_MEMORY_MB);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.SPIN_MILLIS);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.UNIFIER_SINGLE_FINAL);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.IS_OUTPUT_UNIFIED);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.AUTO_RECORD);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.STREAM_CODEC);
+    OUTPUT_ATTRIBUTES.add(Context.PortContext.TUPLE_CLASS);
+
+  }
+
+  /**
+   * The extension point of the stream
+   *
+   * @param <T>
+   */
+  public static class Brick<T>
+  {
+
+    private Operator.OutputPort<T> lastOutput;
+
+    private DagMeta.NodeMeta nodeMeta;
+
+    private Pair<Operator.OutputPort, Operator.InputPort> lastStream;
+
+    public Operator.OutputPort<T> getLastOutput()
+    {
+      return lastOutput;
+    }
+
+    public void setLastOutput(Operator.OutputPort<T> lastOutput)
+    {
+      this.lastOutput = lastOutput;
+    }
+
+    public void setLastStream(Pair<Operator.OutputPort, Operator.InputPort> lastStream)
+    {
+      this.lastStream = lastStream;
+    }
+
+    public Pair<Operator.OutputPort, Operator.InputPort> getLastStream()
+    {
+      return lastStream;
+    }
+  }
+
+  /**
+   * Graph behind the stream
+   */
+  private DagMeta graph;
+
+  private ApexStream<T> delegator;
+
+  /**
+   * Right now the stream only support single extend point
+   * You can have multiple downstream operators connect to this single extend point though
+   */
+  private Brick<T> lastBrick;
+
+  public Brick<T> getLastBrick()
+  {
+    return lastBrick;
+  }
+
+  public void setLastBrick(Brick<T> lastBrick)
+  {
+    this.lastBrick = lastBrick;
+  }
+
+  public ApexStreamImpl()
+  {
+    graph = new DagMeta();
+  }
+
+  public ApexStreamImpl(ApexStream<T> apexStream)
+  {
+    this.delegator = apexStream;
+    if (delegator != null && delegator instanceof ApexStreamImpl) {
+      graph = ((ApexStreamImpl)delegator).graph;
+      lastBrick = ((ApexStreamImpl<T>)delegator).lastBrick;
+    }
+  }
+
+  public ApexStreamImpl(DagMeta graph)
+  {
+    this(graph, null);
+  }
+
+  public ApexStreamImpl(DagMeta graph, Brick<T> lastBrick)
+  {
+    this.graph = graph;
+    this.lastBrick = lastBrick;
+  }
+
+  @Override
+  public <O, STREAM extends ApexStream<O>> STREAM map(Function.MapFunction<T, O> mf)
+  {
+    return map(mf.toString(), mf);
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
```

Comment:
Need to see what causes it. If it's a common call, can it be handled there? Generally I prefer to keep the suppressed section as small as possible and number of occurrences low. Otherwise legit issues can be unintentionally masked.